CONTRIBUTING.md

```diff
@@ -1,10 +1,40 @@
-You want to contribute? We really appreciate that.
-Please have a look at this list to facilitate your contribution:
+# Want to contribute some code? 
+
+## We’d really appreciate your support!
+
+- If the [Github issue](https://github.com/jamulussoftware/jamulus/issues) for your feature/bug fix already exists, write a message in that issue saying you want to work on it. 
+- Otherwise, please [post on the GitHub Discussions](https://github.com/jamulussoftware/jamulus/discussions) and say that you are planning to do some coding and explain why. Then we can discuss the specification. 
+- Please only start coding after we have agreed to a specification.
+
+## Jamulus project/source code general principles
+
+### Stability
+This has been, and must continue to be, the most important requirement. It is particularly important for use cases such as the Worldjam where instability during a live performance would be unacceptable. The following two principles are designed to support this principle.
+
+### [Keep it Simple and Stupid](https://en.wikipedia.org/wiki/KISS_principle) and [Do One Thing and Do It Well](https://en.wikipedia.org/wiki/Unix_philosophy#Do_One_Thing_and_Do_It_Well)
+If a feature or function can be achieved in another way by another system or method, then that is preferable to building that feature into Jamulus. 
+
+### Focus on and improve the strengths of Jamulus
+There are other solutions for real time online jamming available. But Jamulus stands alone with its client/server system, minimalistic GUI and simple functionality. 
+
+### Source code consistency
+- Respect the existing code style: Tab size=4, insert spaces.
+- Space before and after `(` and `)`, except no space between `)` and `;`, and no space before an empty `()`.
+- All bodies of `if`, `else`, `while`, `for`, etc., to be enclosed in braces `{` and `}`, usually on separate lines.
+
+Make sure (if possible) that your code compiles on Windows/Mac/Linux.
+Do not use diff/patch to send your code changes but create a Github fork of the Jamulus code and create a Pull Request when you are done.
+
```

Comment:
I think we should document our compatibility guarantees somewhere and I think this might be the best place?

I don't want to hijack this PR, so if you want to skip this for now, it is ok, then we should have a follow-up PR.

```suggestion
### Supported platforms
We strive to support the following platforms and versions:
- Windows 10 <!-- Is this the only version? Do we have to specify a specific sub-version? -->
- Mac <!-- Which versions? -->
- Ubuntu Linux 18.04 or later, Debian 10 or later, most Linux flavors with recent enough Qt versions
<!-- Do we support BSD? I think a recent discussion was about FreeBSD? -->
<!-- Should we already list Android? If so, what platforms/versions? -->

The minimum Qt version to support is 5.X <-- What is it? -->

Please try hard to avoid breaking any of them by avoiding to introduce platform-specific code and checking if any newly introduced Qt calls are available in our minimum Qt version.

### Dependencies
If your code requires new dependencies, be extra sure that those are available on all supported platforms and that the introduction of those has been discussed and accepted.
```

Reply:
Windows 7 was what I thought we aimed for as a minimum version.  Later should "just work" with ASIO and Qt:
https://doc.qt.io/qt-5/windows.html

Basically, I think cross-platform, it's "Whatever Qt5 and our supported sound platform is support".

Reply:
> Windows 7 was what I thought we aimed for as a minimum version. Later should "just work" with ASIO and Qt:
> https://doc.qt.io/qt-5/windows.html

Yes, I agree. I still use Windows 7 on my main PC, which is nearly 9 years old. I think we would need to have a good reason to stop working on W7.

> I think some (Mac?) platforms needed Qt5.9 for some reason...

Apparently Qt 5.9 supports back to macOS 10.10 (Yosemite), 5.10/5.11 support back to macOS 10.11 (El Capitan), 5.12/5.13 support back to macOS 10.12 (Sierra) and 5.14/5.15 support back to macOS 10.13 (High Sierra). We have no way of knowing how many users need support for those old versions. I use 10.14 (Mojave) on mine.

Reply:
The reason for listing explicit (minimum) OS versions was to find out what the oldest supported Qt version is so that we can list it. If we just specify a minimum Qt version, it may be hard to raise that at some point.

> I think we need to start working on a Qt exit plan...

Should we have a Github discussion about that? I've only recently learned about these changes in Qt.